sdks/unity/AgonesSdk.cs

```diff
@@ -183,9 +183,34 @@ public async Task<bool> SetAnnotation(string key, string value)
             return await SendRequestAsync("/metadata/annotation", json, UnityWebRequest.kHttpVerbPUT)
                 .ContinueWith(task => task.Result.ok);
         }
+
+        private struct Duration
+        {
+            public int seconds;
+
+            public Duration(int seconds)
+            {
+                this.seconds = seconds;
+            }
+        }
+
+        /// <summary>
+        /// Move the GameServer into the Reserved state for the specified Timespan (0 seconds is forever)
+        /// Smallest unit is seconds.
+        /// </summary>
+        /// <param name="duration">The time span to reserve for</param>
+        /// <returns>
+        /// A task that represents the asynchronous operation and returns true if the request was successful
+        /// </returns>
+        public async Task<bool> Reserve(TimeSpan duration)
+        {
+            string json = JsonUtility.ToJson(new Duration(seconds: duration.Seconds));

```

Comment:
duration.Duration() instead of new Duration()?
https://docs.microsoft.com/en-us/dotnet/api/system.timespan.duration?view=netframework-4.8

Reply:
This is a `Duration` struct as defined above, I assume to match the expected JSON format of the request.

Reply:
Yes, that is indeed the reason.

Unless there is a good reason to grab the absolute value of the timespan?